systems/plants/RigidBodyFlatTerrain.m

```diff
@@ -3,13 +3,18 @@
 %  everywhere
   
   methods 
-    function obj = RigidBodyFlatTerrain()
+    function obj = RigidBodyFlatTerrain(varargin)
+      % Construct a flat terrain map at a given z height (default 0)
+      p = inputParser();
```

Comment:
Looks like `inputParser` is [slower for checking optional inputs](http://stackoverflow.com/a/26635564/727415), but I haven't found anything on the kind of field checks that we do on our options structs. I'm guessing that we wont want to use it in 'inner-loop' code, but I'd argue that we probably shouldn't be parsing options structs there either.

I tried converting a couple of functions over to the `inputParser` style last night - I think I like it better.